syntax = "proto3";

package smart.co;

import "google/protobuf/empty.proto";

message SensorState {
  bool is_on = 1;
  float co_level = 2;
}

message Ack {
  string message = 1;
}

service COSensor {
  rpc GetState (google.protobuf.Empty) returns (SensorState);
  rpc ToggleSensor (google.protobuf.Empty) returns (Ack);
  rpc ReadCOLevel (google.protobuf.Empty) returns (SensorState);
}
